### ISSUE TITLE
Timeout Errors

### ISSUE THREAD

**Issue by MinniArora** [eclipse-vertx/vert.x#3399]
### Version

3.4.0

### Context

We have few services exposed over vertx and there is a read timeout of 1s. 

Whenever a service becomes slow due to issues on the backend, we get timeout errors which is expected: The error message is: The timeout period of 1000ms has been exceeded while executing GET .... for host ...

But even after the service recovers, we keep getting timeout errors continuously and now the requests do not even reach the service. Seems the requests are timing out from some internal queue. It's only after a restart that the issue gets resolved.

Stacktrace:
        at io.vertx.core.http.impl.HttpClientRequestBase.handleException(HttpClientRequestBase.java:136)
        at io.vertx.core.http.impl.HttpClientRequestImpl.handleException(HttpClientRequestImpl.java:51)
        at io.vertx.core.http.impl.HttpClientRequestBase.timeout(HttpClientRequestBase.java:183)
        at io.vertx.core.http.impl.HttpClientRequestBase.handleTimeout(HttpClientRequestBase.java:168)
        at io.vertx.core.http.impl.HttpClientRequestBase.lambda$setTimeout$0(HttpClientRequestBase.java:126)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:784)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:755)
        at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:324)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:445)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
        at java.lang.Thread.run(Thread.java:748)

### Extra

Basically, we are using apiman (https://github.com/apiman/apiman) as an api gateway to expose services and it is built on top of Vertx

**Comment by vietj** [comment]
Hi,

I would advise to use a more recent version of Vert.x to check if that still happens ?

Julien

**Comment by MinniArora** [comment]
Thank you for a quick reply.

Checked version 3.7.0 and the same problem exists.

**Comment by vietj** [comment]
can you try with very recent versions like 3.8 or 3.9 ?

can you provide a reproducer ?

it seems to me that your client might have very short timeout time and lot of queuing might happen in an under sized pool.

have you tried increasing pool size ?

**Comment by MinniArora** [comment]
Sorry, cannot upgrade at this point. Could you please share the variables that can be used to increase the pool size.
